app/domain/authentication/authn_k8s/extract_k8s_resource_restrictions.rb

```diff
@@ -0,0 +1,71 @@
+require 'command_class'
+
+module Authentication
+  module AuthnK8s
+
+    ExtractK8sResourceRestrictions = CommandClass.new(
+      dependencies: {
+        extract_resource_restrictions: ResourceRestrictions::ExtractResourceRestrictions.new,
+        resource_restrictions_class:   ResourceRestrictions::ResourceRestrictions,
+        logger:                        Rails.logger
+      },
+      inputs:   %i(authenticator_name service_id role_name account)
+    ) do
+
+      def call
+        extract_resource_restrictions
+      end
+
+      private
+
+      # Extracts resource restrictions for k8s hosts. It can be defined in the
+      # annotations or as part of the host ID itself.
+      # If the restrictions are in the annotations, then they will not be
+      # extracted from the host ID.
+      def extract_resource_restrictions
+        get_restrictions_from_annotations = @extract_resource_restrictions.call(
```

Comment:
this should be `restrictions_from_annotations`, no? When i see an object named `get_restrictions_from_annotations` i think it is an instance of a function.

Reply:
Correct, my bad... I copied Jonah's suggestion without noticing this.
Fixed